File: frontend/src/services/ComplaintService.js
import api from "./api";

export const raiseComplaint = async (data) => {
    const response = await api.post("/complaints", data);
    return response.data;
};

export const getMyComplaints = async () => {
    const response = await api.get("/complaints/my");
    return response.data;
};

export const getComplaintById = async (id) => {
    const response = await api.get(`/complaints/${id}`);
    return response.data;
};

export const getCommunityComplaints = async () => {
    const response = await api.get("/complaints/community");
    return response.data;
};

export const searchComplaints = async (params) => {
    const response = await api.get("/complaints/search", { params });
    return response.data;
};

export const updateComplaint = async (id, data) => {
    const response = await api.put(`/complaints/${id}`, data);
    return response.data;
};

File: monitoring-and-billing-platform/src/main/java/com/water/monitoring_and_billing_platform/service/ComplaintService.java
package com.water.monitoring_and_billing_platform.service;

import com.water.monitoring_and_billing_platform.dto.ComplaintRequest;
import com.water.monitoring_and_billing_platform.dto.ComplaintResponse;
import com.water.monitoring_and_billing_platform.enums.ComplaintCategory;
import com.water.monitoring_and_billing_platform.enums.ComplaintPriority;
import com.water.monitoring_and_billing_platform.enums.ComplaintStatus;

import java.util.List;

public interface ComplaintService {
    ComplaintResponse raiseComplaint(String email, ComplaintRequest request);
    List<ComplaintResponse> getMyComplaints(String email);
    ComplaintResponse getComplaintById(String email, Long id);
    List<ComplaintResponse> getCommunityComplaints(String email);
    List<ComplaintResponse> searchCommunityComplaints(
            String email,
            ComplaintStatus status,
            ComplaintPriority priority,
            ComplaintCategory category,
            String search
    );
    ComplaintResponse updateComplaint(
            String email,
            Long id,
            ComplaintStatus status,
            String remarks,
            Long assignedToUserId
    );
}
